discovery/pkg/openstack/reconciler.go

```diff
@@ -189,3 +192,31 @@ func (r *Reconciler) reconcileEndpoints(desired, current []v1.Endpoints) {
 		r.syncqueue.Enqueue(sync.DeleteEndpointsAction(&ep))
 	}
 }
+
+func (r *Reconciler) skipInvalidLoadBalancers(projectName string, lbs []loadbalancers.LoadBalancer) []loadbalancers.LoadBalancer {
+	// Kubernetes DNS_LABEL, but allows uppercase letters
+	validName := regexp.MustCompile("^[a-zA-Z]([-a-zA-Z0-9]*[a-zA-Z0-9])?$")
+	valid := []loadbalancers.LoadBalancer{}
+
+	for _, lb := range lbs {
+		if lb.Name != "" && !validName.MatchString(lb.Name) {
+			r.Metrics.GenericMetricError(r.ClusterName, "InvalidLoadBalancerName")
```

Comment:
Should this be a `service` error rather than a generic one?

Reply:
Just looked into using a service error, but we don't have a service name at this point, given that it is invalid.